diff --git a/src/gpu/ganesh/mtl/GrMtlCaps.mm b/src/gpu/ganesh/mtl/GrMtlCaps.mm
index 8d0bc6e364..26c79df714 100644
--- a/src/gpu/ganesh/mtl/GrMtlCaps.mm
+++ b/src/gpu/ganesh/mtl/GrMtlCaps.mm
@@ -102,8 +102,10 @@ bool GrMtlCaps::getGPUFamilyFromFeatureSet(id<MTLDevice> device,
             return true;
         }
     }
-#elif defined(SK_BUILD_FOR_IOS)
-    // TODO: support tvOS
+#endif
+
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
+#if defined(SK_BUILD_FOR_IOS)
    *gpuFamily = GPUFamily::kApple;
     // iOS 12
     if (@available(iOS 12.0, tvOS 12.0, *)) {
@@ -164,6 +166,35 @@ bool GrMtlCaps::getGPUFamilyFromFeatureSet(id<MTLDevice> device,
     }
     // We don't support earlier OSes
 #endif
+#endif  // !(defined(TARGET_OS_TV) && TARGET_OS_TV)
+
+#if defined(TARGET_OS_TV) && TARGET_OS_TV
+    *gpuFamily = GPUFamily::kApple;
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily2_v2]) {
+        *group = 2;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily2_v1]) {
+        *group = 2;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v4]) {
+        *group = 1;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v3]) {
+        *group = 1;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v2]) {
+        *group = 1;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v1]) {
+        *group = 1;
+        return true;
+    }
+#endif
 
 #endif // GR_METAL_SDK_VERSION < 300
 
@@ -172,6 +203,10 @@ bool GrMtlCaps::getGPUFamilyFromFeatureSet(id<MTLDevice> device,
 }
 
 bool GrMtlCaps::getGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* group) {
+#if defined(TARGET_OS_TV) && TARGET_OS_TV
+    // Apple6/7 family not available on tvOS; use getGPUFamilyFromFeatureSet instead.
+    return false;
+#else
 #if GR_METAL_SDK_VERSION >= 220
     if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
         // Apple Silicon
@@ -236,6 +271,7 @@ bool GrMtlCaps::getGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* gr
 
     // No supported GPU families were found
     return false;
+#endif  // !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 }
 
 void GrMtlCaps::initGPUFamily(id<MTLDevice> device) {
diff --git a/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm b/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
index 3bd7a192c1..5643a4f965 100644
--- a/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
+++ b/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
@@ -23,6 +23,9 @@ GR_NORETAIN_BEGIN
 
 sk_sp<GrMtlCommandBuffer> GrMtlCommandBuffer::Make(id<MTLCommandQueue> queue) {
     id<MTLCommandBuffer> mtlCommandBuffer;
+#if (defined(TARGET_OS_TV) && TARGET_OS_TV)
+    mtlCommandBuffer = [queue commandBuffer];
+#else
 #if GR_METAL_SDK_VERSION >= 230
     if (@available(macOS 11.0, iOS 14.0, tvOS 14.0, *)) {
         MTLCommandBufferDescriptor* desc = [[MTLCommandBufferDescriptor alloc] init];
@@ -34,6 +37,7 @@ sk_sp<GrMtlCommandBuffer> GrMtlCommandBuffer::Make(id<MTLCommandQueue> queue) {
 #else
     mtlCommandBuffer = [queue commandBuffer];
 #endif
+#endif  // TARGET_OS_TV
     if (nil == mtlCommandBuffer) {
         return nullptr;
     }
diff --git a/src/gpu/ganesh/mtl/GrMtlGpu.mm b/src/gpu/ganesh/mtl/GrMtlGpu.mm
index 0cac6cb18e..31b1437a07 100644
--- a/src/gpu/ganesh/mtl/GrMtlGpu.mm
+++ b/src/gpu/ganesh/mtl/GrMtlGpu.mm
@@ -1814,11 +1814,13 @@ void GrMtlGpu::onDumpJSON(SkJSONWriter* writer) const {
         writer->appendU64("maxTransferRate", fDevice.maxTransferRate);
     }
 #endif  // SK_BUILD_FOR_MAC
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500 || __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
     if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
         writer->appendBool("hasUnifiedMemory", fDevice.hasUnifiedMemory);
     }
 #endif
+#endif  // !TARGET_OS_TV
 #ifdef SK_BUILD_FOR_MAC
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500
     if (@available(macOS 10.15, *)) {
@@ -1898,11 +1900,13 @@ void GrMtlGpu::onDumpJSON(SkJSONWriter* writer) const {
     if (@available(macOS 10.14, iOS 12.0, tvOS 12.0, *)) {
         writer->appendU64("maxArgumentBufferSamplerCount", fDevice.maxArgumentBufferSamplerCount);
     }
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 #ifdef SK_BUILD_FOR_IOS
     if (@available(iOS 13.0, tvOS 13.0, *)) {
         writer->appendU64("sparseTileSizeInBytes", fDevice.sparseTileSizeInBytes);
     }
 #endif
+#endif  // !TARGET_OS_TV
     writer->endObject();
 
     writer->appendCString("queue", fQueue.label.UTF8String);
diff --git a/src/gpu/graphite/mtl/MtlCaps.mm b/src/gpu/graphite/mtl/MtlCaps.mm
index 871575ff10..c3b9b16331 100644
--- a/src/gpu/graphite/mtl/MtlCaps.mm
+++ b/src/gpu/graphite/mtl/MtlCaps.mm
@@ -40,6 +40,7 @@ MtlCaps::MtlCaps(const id<MTLDevice> device, const ContextOptions& options)
 }
 
 bool MtlCaps::GetGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* group) {
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 #if SKGPU_GRAPHITE_METAL_SDK_VERSION >= 220
     if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
         // Apple Silicon
@@ -101,6 +102,7 @@ bool MtlCaps::GetGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* grou
         }
     }
 #endif
+#endif  // !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 
     // No supported GPU families were found
     return false;
